server/utils/passport/ensure-authenticated.js

```diff
@@ -1,33 +1,24 @@
-import basePathConfig from '../../configs/basePathConfig'
+import { routes } from '../routes'
 
-const basePath = basePathConfig || ''
-const logoutForbiddenRoute = `${basePath}/logout?e=forbidden`
-const unauthorizedRoute = `${basePath}/logout?e=unauthorized`
-
-export default function ensureAuthenticated(req, res, next) {
-  if (!req.isAuthenticated()) {
-    return res.redirect(`${basePath}/logout`)
+export default async function ensureAuthenticated(req, res, next) {
+  try {
+    if (!req.isAuthenticated()) {
+      return res.redirect(routes.logout)
+    }
+    const { prisma } = req.app.locals
+    const user = await prisma.users.findFirst({
+      where: { uid: req.user },
+      select: { access: true, blocked: true },
+    })
+    switch (true) {
+      case !user || Object.keys(user).length === 0 || !!user.blocked:
+        return res.redirect(routes.logout)
+      case user.access && user.access.length === 0:
```

Comment:
We are now defaulting access to an empty array so we shouldn't have to check it here.

```suggestion
      case !user.access.length:
```

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/2e3f2e5a20221046bc208d7918023e42684da6e2

Reply:
That commit shows a change here but the overall diff does not 🤔

Reply:
The commit is part of this pr https://github.com/AMP-SCZ/dpdash/pull/241